### PR TITLE
snap: add demo-server

### PR DESCRIPTION
**Pull request by Saviq**
This will ease testing in the lab

### CODE REVIEW

File: wrapper/server-wrapper [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+export XDG_RUNTIME_DIR=$( dirname $XDG_RUNTIME_DIR )
+
```
Comment (AlanGriffiths) [inline wrapper/server-wrapper:3]: Maybe want to add something like 

```bash
if [ -n "${WAYLAND_DISPLAY}" ] && [ -O "${XDG_RUNTIME_DIR}/${WAYLAND_DISPLAY}" ]
then
  echo "ERROR: WAYLAND_DISPLAY is set, and already provided by a compositor. Mir will be unable to start!"
  exit 1
fi
```
Or 
```bash
if [ -n "${WAYLAND_DISPLAY}" ] && [ -O "${XDG_RUNTIME_DIR}/${WAYLAND_DISPLAY}" ]
then
  echo "Info: wayland endpoint '${WAYLAND_DISPLAY}' already exists, using it as host"
  export MIR_SERVER_WAYLAND_HOST="${WAYLAND_DISPLAY}"
  unset WAYLAND_DISPLAY
fi
```

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]
I think the behaviour when running under a Wayland desktop leaves something to be desired.

The simplest approach would be to unset `WAYLAND_DISPLAY` in `server-wrapper` which would, in this case, allow mir_demo_server to start on X11.

(See other comment for less simple options)

**Comment by Saviq** [comment]
> The simplest approach would be to unset `WAYLAND_DISPLAY` in `server-wrapper` which would, in this case, allow mir_demo_server to start on X11.
> 
> (See other comment for less simple options)

Will do, I do want `WAYLAND_DISPLAY` to be respected, though - so will go with one of your suggestions.

**Review by AlanGriffiths — APPROVED** [review]
bors merge

**Pull request merged**